File: DevLab/Python/Official/number1.py
#https://borntodev.com/devlab/task/551

def reduce_to_single_digit(num):
    while num > 9:
        num = sum(int(digit) for digit in str(num))
    return num

    
input_str = input()
    
total_sum = 0
    
for char in input_str:
    char_val = ord(char)
    powered_val = char_val ** len(input_str)
    total_sum += powered_val
    
result = reduce_to_single_digit(total_sum)
    
print(result)

File: DevLab/Java/Official/number1.java
//https://borntodev.com/devlab/task/551

import java.util.Scanner;

public class number1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        sc.close();

        long sum = 0;
        for (char c : input.toCharArray()) {
            int char_val = (int) c;
            long powered_val = (long) Math.pow(char_val, input.length());
            sum += powered_val;
        }

        while(sum >= 10){
            sum = final_sum(sum);
        }
        
        System.out.println(sum);
    }
    static long final_sum(long num) {
        long sum = 0;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
}
